src/Catalyst.Core.Modules.Dfs/UnixFs/ChunkedStream.cs

```diff
@@ -0,0 +1,152 @@
+                });
+                position += dm.BlockSizes[i];
+            }
+        }
+
+        IBlockApi BlockService { get; set; }
+        IKeyApi KeyChain { get; set; }
+
+        /// <inheritdoc />
+        public override long Length => fileSize;
+
+        /// <inheritdoc />
+        public override void SetLength(long value) { throw new NotSupportedException(); }
+
+        /// <inheritdoc />
+        public override bool CanRead => true;
+
+        /// <inheritdoc />
+        public override bool CanSeek => true;
+
+        /// <inheritdoc />
+        public override bool CanWrite => false;
+
+        /// <inheritdoc />
+        public override void Flush() { }
+
+        /// <inheritdoc />
+        public override long Position { get; set; }
+
+        /// <inheritdoc />
+        public override long Seek(long offset, SeekOrigin origin)
+        {
+            switch (origin)
+            {
+                case SeekOrigin.Begin:
+                    Position = offset;
+                    break;
+                case SeekOrigin.Current:
+                    Position += offset;
+                    break;
+                case SeekOrigin.End:
+                    Position = Length - offset;
+                    break;
+            }
+
+            return Position;
+        }
+
+        /// <inheritdoc />
+        public override void Write(byte[] buffer, int offset, int count) { throw new NotSupportedException(); }
+
+        /// <inheritdoc />
+        public override int Read(byte[] buffer, int offset, int count)
+        {
+            return ReadAsync(buffer, offset, count).ConfigureAwait(false).GetAwaiter().GetResult();
+        }
+
+        /// <inheritdoc />
+        public override async Task<int> ReadAsync(byte[] buffer, int offset, int count, CancellationToken cancel)
+        {
+            var block = await GetBlockAsync(Position, cancel).ConfigureAwait(false);
+            var k = Math.Min(count, block.Count);
+            if (k > 0)
+            {
+                Array.Copy(block.Array, block.Offset, buffer, offset, k);
+                Position += k;
+            }
+
+            return k;
+        }
+
+        BlockInfo currentBlock;
+        byte[] currentData;
+
+        async Task<ArraySegment<byte>> GetBlockAsync(long position, CancellationToken cancel)
+        {
+            if (position >= Length)
+            {
+                return new ArraySegment<byte>();
+            }
+
+            var need = blocks.Last(b => b.Position <= position);
+            if (need != currentBlock)
+            {
+                var stream = await UnixFs.CreateReadStreamAsync(need.Id, BlockService, KeyChain, cancel)
+                   .ConfigureAwait(false);
+                currentBlock = need;
+                currentData = new byte[stream.Length];
+                for (int i = 0, n; i < stream.Length; i += n)
+                {
+                    n = await stream.ReadAsync(currentData, i, (int) stream.Length - i);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Do not update the loop counter 'n' within the loop body.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4612924)